src/builder/webpack/config.common.js

```diff
@@ -41,4 +38,17 @@ export default (config: PhenomicConfig): WebpackConfig => {
       } }),
     ],
   }
+
+  if (config.cache) {
+    return {
+      ...defaultConfig,
+      ...hardSourceRecordsPath(cacheDir),
+      plugins: [
+        ...defaultConfig.plugins,
+        hardSourcePlugin(cacheDir, config),
+      ],
+    }
+  }
+
+  return defaultConfig
```

Comment:
Can you avoid this big block and instead do something like `...config.cache ? hardSourceRecordsPath(cacheDir) : {},`?